cueweb/README.md

```diff
@@ -0,0 +1,324 @@
+- CueWeb is designed to be platform-independent, functioning seamlessly on macOS, Windows, and Linux. There are no specific OS version requirements.
+
+**Dependency versions:**
+- For precise version information on all dependencies, consult the `package.json` file located in the project repository. This resource ensures you have the correct versions of the required libraries to run CueWeb effectively.
+
+**Docker support:**
+- If you prefer a containerized environment, ensure [Docker](https://docs.docker.com/get-docker) is installed to facilitate the deployment of CueWeb.
+
+Go back to [Contents](#contents).
+
+# Installation and Usage
+
+Next is the process to install and use the CueWeb system.
+
+## Step-by-Step Instructions
+
+### Setting up environment variables
+
+- To run CueWeb, certain environment variables must be set for the application to run as intended. Set up the following environment variables, either by adding them to the docker file directly (`ENV MY_ENV_VARIABLE="value"`), or adding a `.env` file in `/cueweb` where these variables are configured (if you do the latter, make sure you copy in that file in the docker file i.e: `COPY .env /opt/cueweb/.env`). If your application will be deployed on Openshift, those environment variables can be configured there as well.
+    - NEXT_PUBLIC_OPENCUE_ENDPOINT
+        - This is a gateway that provides a REST endpoint to the opencue gRPC API. This is needed for jobs, layers, and frames to be retrieved. See [Opencue REST Gateway](https://github.com/AcademySoftwareFoundation/OpenCue/blob/master/rest_gateway/README.md).
+        - This means that this rest gateway must be set-up and the environment variable should be set to the url for rest gateway.
+    - NEXT_PUBLIC_URL
+        - This is the URL where CueWeb is hosted and accessible by users ex: localhost:3000
+    - NEXT_JWT_SECRET
+        - This is used to create a JWT token which is required to access the REST endpoint of the opencue gRPC API
+
+    - Authentication environment variables
+        - Depending on which provider you use for authentication, you may have to set certain environment variables. 
+        - For example, for authentication (see file `cueweb/lib/auth.ts`), the following environment variables must be set:
+            - `NEXT_PUBLIC_AUTH_PROVIDER=github,okta,google` will show the three authentication buttons (Okta, Google, GitHub).
+            - `NEXT_PUBLIC_AUTH_PROVIDER=github,google` will show the two authentication buttons (GitHub and Google).
+            - `NEXT_PUBLIC_AUTH_PROVIDER=google` will show only the Google (Gmail) authentication button.
+
+          - [Okta](https://www.okta.com/)
+              - NEXT_AUTH_OKTA_CLIENT_ID
+              - NEXT_AUTH_OKTA_ISSUER
+              - NEXT_AUTH_OKTA_CLIENT_SECRET
+          - Google (Gmail)
+            - GOOGLE_CLIENT_ID
+            - GOOGLE_CLIENT_SECRET
+          - GitHub
+            - GITHUB_ID
+            - GITHUB_SECRET
```

Comment:
As mentioned on yesterday's meeting. For some studios, having authentication can be a blocker. I think this section should explain how to turn authentication off.

Reply:
Done!

See commit: [cueweb] Make CueWeb authentication and Sentry usage optional
https://github.com/AcademySoftwareFoundation/OpenCue/pull/1596/commits/9c326b6b4e072563281fa3a92ac8cbd8c93f6ef3

- Added functionality to make login and Sentry usage optional in CueWeb.
- Updated configuration to enable/disable these features via environment variables.
- Enhanced the cueweb/README.md with instructions for configuring these options in the CueWeb setup.

### About CueWeb authentication usage

- Authentication environment variables
   - Depending on which [Next-Auth.js](https://next-auth.js.org/) provider you use for authentication, you may have to set certain environment variables. 
   - For example, for authentication (see file `cueweb/lib/auth.ts`), the following environment variables must be set:
      - `NEXT_PUBLIC_AUTH_PROVIDER=github,okta,google` will show the three authentication buttons (Okta, Google, GitHub).
      - `NEXT_PUBLIC_AUTH_PROVIDER=github,google` will show the two authentication buttons (GitHub and Google).
      - `NEXT_PUBLIC_AUTH_PROVIDER=google` will show only the Google (Gmail) authentication button.

      - [Okta](https://www.okta.com/)
            - NEXT_AUTH_OKTA_CLIENT_ID
            - NEXT_AUTH_OKTA_ISSUER
            - NEXT_AUTH_OKTA_CLIENT_SECRET
      - Google (Gmail)
            - GOOGLE_CLIENT_ID
            - GOOGLE_CLIENT_SECRET
      - GitHub
            - GITHUB_ID
            - GITHUB_SECRET

- To disable the CueWeb authentication, do not define `NEXT_PUBLIC_AUTH_PROVIDER`.

Change `.env`

From:

```env
# Authentication Configuration:
NEXT_PUBLIC_AUTH_PROVIDER=github,okta,google
```

To:

```env
# Authentication Configuration:
# NEXT_PUBLIC_AUTH_PROVIDER=github,okta,google
```

#### Authentication Setup 

The CueWeb project utilizes the [NextAuth.js](https://next-auth.js.org/) library for authentication, which includes many popular providers out-of-the-box for additional configuration or for implementing your own email authentication with a custom database. This project already implements Google, Github and Okta authentication and they are enabled if their respective environment variables are provided, otherwise ignored. 

Authentication providers in [NextAuth.js](https://next-auth.js.org/) are services that can be used to sign in to a user.

There are four ways a user can be signed in:

- [Using a built-in OAuth Provider](https://next-auth.js.org/configuration/providers/oauth) (e.g Github, Google, Okta, Apple, GitLab, Amazon, Microsoft Azure, LinkedIn, Atlassian, Auth0, etc...)
   - List of NextAuth.js providers: https://next-auth.js.org/providers/.
- [Using a custom OAuth Provider](https://next-auth.js.org/configuration/providers/oauth#using-a-custom-provider)
- [Using Email](https://next-auth.js.org/configuration/providers/email)
- [Using Credentials](https://next-auth.js.org/configuration/providers/credentials)